20250827 fix zugferd require and errors

bblessmann opened the pull request:
Ein require im ZUGFeRD-DB-Helper hat eine nicht vorhandene Manager-Klasse laden wollen.
Zudem haben die evals beim Aufruf der entsprechenden Methode in io.pl die Fehler verschluckt, so dass außer einem internal server error kein Hinweis darauf kam.
Hier ein Fix dazu.

bblessmann committed:
ZUGFeRD: require für S:D:GenericTranslation, nicht für Manager davon.

Die Manager-Klasse hat keine eigene Modul-Datei.

bblessmann committed:
ZUGFeRD-Export: andere Fehler als von der Validierung nicht verschlucken.

Sonst gibt es nur einen "internal server error" und die Benutzerin kann der
Entwicklerin keinen Hinweis auf den Fehler liefern.

sschoeling commented on bin/mozilla/io.pl:
```diff
@@ -2359,7 +2359,7 @@ sub _maybe_attach_zugferd_data {
       }
     ];
-  };
+  } or do { $::form->error($@) };
```
Meh, da musst du dann aber auch das "if SL::X::ZugGeRDValidation->caught" anpassen in beiden Fällen. Fände ich auch ehrlich gesagt besser, weil ich das Pattern was hier im Moment ist garnicht mag.

bblessmann committed:
ZUGFeRD-Export: toten Code entfernt

Da alle Ausnhamen im eval-Block gefangen werden, ist die extra Behandlung
von Validierungs-Ausnahman danach sinnlos.
